server/listeners/eventHandlers/prisonEvents/datesChangedEventHandler.ts

```diff
@@ -5,11 +5,16 @@ import { PrisonEventMessage } from '../../../@types/prisonApiClientTypes'
 import LicenceStatus from '../../../enumeration/licenceStatus'
 import { convertDateFormat } from '../../../utils/utils'
 
-export default class SentenceDatesChangedEventHandler {
+export default class DatesChangedEventHandler {
   constructor(private readonly licenceService: LicenceService, private readonly prisonerService: PrisonerService) {}
 
   handle = async (event: PrisonEventMessage): Promise<void> => {
-    const nomisId = event.offenderIdDisplay
+    const { bookingId, offenderIdDisplay } = event
+
+    const nomisId =
+      offenderIdDisplay ||
+      (await this.prisonerService.searchPrisonersByBookingIds([bookingId])).map(o => o.prisonerNumber).pop()
+
```

Comment:
The RELEASE_DATE_CHANGED event only provides a bookingId, where as the SENTENCE_DATES-CHANGED event provides a nomisId ... _*sigh*_

So if a nomisId exists, use that, and if not, then use the bookingId to find the nomisId